src/components/toolbar/toolbar-button-manager.tsx

```diff
@@ -0,0 +1,66 @@
+import { ITileModel } from "../../models/tiles/tile-model";
+
+// The toolbar button manager registers all known buttons used on tile toolbars.
+// Tiles and plugins can register buttons by specifying their basic attributes
+// and the Component to render for the button.
+// These buttons can make use of some basic information passed in to all buttons
+// (eg, the tile model) or additional information provided through a Context in the tile.
+// Each toolbar has a default set of buttons, but which buttons are actually rendered,
+// and in what order, can be customized by the unit, lesson, and problem content configurations.
+
+export interface IToolbarButtonInfo {
+  name: string,  // a unique named used in configuration to identify the button
+  title: string, // user-visible tooltip for the button
+  component: React.ComponentType<IToolbarButtonProps>, // component to render
+  keyHint?: string, // If set, displayed to the user as the hotkey equivalent
+  defaultPosition?: number, // If set, shown on the tile's default toolbar in this position
```

Comment:
That makes sense, I hadn't considered how to add a `['dataset-view', 'Table']` to the default set.

Alternatively, could there be a global default 'store' so that `useSettingFromStores` would just return that default if there's nothing more specific, and a separate place for defaults is not needed?